substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVLineRecordBuilder.java

```diff
@@ -66,9 +67,9 @@ CVLineRecord build(CompiledMethodEntry entry) {
         debug("CVLineRecord.computeContents: processing primary range %s", primaryRange);
 
         processRange(primaryRange);
-        Iterator<Range> iterator = compiledEntry.leafRangeIterator();
+        Iterator<SubRange> iterator = compiledEntry.leafRangeIterator();
```

Comment:
Well, perhaps but I don't see any reason to just now. These subclasses are not currently exposed outside of the type hierarchy (they are declared as protected) because their existence is merely an implementation detail. They both implement the same API, defined at the level of SubRange. So, even if a client uses what can only be a LeafRange via this API it cannot invoke a behaviour that is only available for a LeafRange.
What I can do though is simplify some of the behaviours that are still distributed across all Range subclasses e.g. A Primary does not need to store local values Method includesInlineRanges can be defined once in Range, rather than in three different versions in Primary, Call and Leaf.
I'll do that in the next reworking.